pyout.py

```diff
@@ -540,6 +545,18 @@ def __init__(self, columns=None, style=None, stream=None, force_styling=False):
             self._setup_style()
             self._setup_fields()
 
+        self._pool = None
+        self._lock = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        if self._pool is None:
+            return
+        self._pool.close()
+        self._pool.join()
```

Comment:
I find it more "flexible" to support the classical/procedural `wait()` (which you showed I believe) and then just use it in the `__exit__`. This way programmer could use context manager as well as "create/use/wait"   sequence

Reply:
fair enough